src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -147,7 +148,14 @@ protected function setDefinition($id, Definition $definition)
             }
             $this->instanceof[$id] = $definition;
         } else {
-            $this->container->setDefinition($id, $definition instanceof ChildDefinition ? $definition : $definition->setInstanceofConditionals($this->instanceof));
+            $conditionals = array_filter($this->instanceof, function (string $instanceOfId) use ($id) {
+                if (!isset($this->instanceofIgnored[$instanceOfId])) {
```

Comment:
no need for this "if" block, the next condition is enough as it embeds this check already

Reply:
I was wondering whether `isset($this->instanceofIgnored[$instanceOfId])` could throw a warning if `$instanceOfId` is not presented in the array or not. Done!